core/liblwm2m.h

```diff
@@ -332,10 +324,44 @@ typedef enum
     BINDING_UQS  // UDP queue mode plus SMS
 } lwm2m_binding_t;
 
+typedef enum    // data type of TLV_RESSOURCE for cond. observe
+{
+    LWM2M_DATATYPE_UNKNOWN = 0,
+    LWM2M_DATATYPE_STRING,
+    LWM2M_DATATYPE_INTEGER,
+    LWM2M_DATATYPE_FLOAT,
+    LWM2M_DATATYPE_BOOLEAN,
+    LWM2M_DATATYPE_OPAQUE,
+    LWM2M_DATATYPE_TIME
+} lwm2m_data_type_t;
+
```

Comment:
This could be added to the lwm2m_tlv_t stucture. So no need for a new object callback.